app/src/main/java/com/nervousfish/nervousfish/data_objects/RSAKey.java

```diff
@@ -36,24 +34,46 @@ public RSAKey(final String name, final String modulus, final String exponent) {
     }
 
     /**
-     * Create a new RSAKey given a {@link JsonReader}.
+     * Constructor for a RSA key given a {@link Map} of its values.
      *
-     * @param reader The {@link JsonReader} to read with.
-     * @return An {@link IKey} representing the JSON key.
-     * @throws IOException When the {@link JsonReader} throws an {@link IOException}.
+     * @param map A {@link Map} mapping {@link RSAKey} attribute names to values.
      */
-    static public IKey fromJSON(final JsonReader reader) throws IOException {
-        final Map<String, String> map = new ConcurrentHashMap<>();
-        while (reader.hasNext()) {
-            final String name = reader.nextName();
-            final String value = reader.nextString();
-            map.put(name, value);
+    public RSAKey(final Map<String, String> map) {
```

Comment:
Can you check if the map contains these strings and otherwise log it or even throw an exception?